### ISSUE TITLE
deleting an image from a "bildschirm" doesn't work

### ISSUE THREAD

**Issue by GVogeler** [cceh/palonline#15]
If I delete the `shockwave.jpg` entry in the `Bildteil` field in http://www.palaeographie-online.de/vhb_intern/form.php?tab=bildschirm&ID=172 the value stays in the record.
The same, if I additionally set the select box of the `Bildteil` to '... Keine Zuordnung'.

**Comment by olvidalo** [comment]
This happens because both of the `bildteil` text fields, the one before and the one after the = sign, have the same name field_Bildteil. Thus, the following post data is sent to write.php

```
field_Bildteil:
field_Bildteil:shockwave.jpg
```

and the one where there is an actual value apparently gets precedence when php parses it.

Which of the values should be actually written to the database into the `Bildteil` column?

**Comment by GVogeler** [comment]
If we have an upload mechism for images we could skip the text field before the =.

But I wonder why emptying the text field and setting the select field to "... Keine Zuordnung" has the same behaviour of "not deleting"?

**Comment by olvidalo** [comment]
The "...Keine Zuordnung.." option-Tag has an empty `value` attribute which is equivalent for value="". So when this is selected, the following post data is sent:

```
field_Bildteil:shockwave.jpg
field_Bildteil:
```

where, again, the value with a value takes precedence.

**Comment by GVogeler** [comment]
Thus but emptying both should do the work - but doesn't.

Am 10.04.2015 um 15:14 schrieb Marcel:

**Comment by olvidalo** [comment]
You're right. The problem is in `write.php`` in line 72, where it checks for empty values and only adds non-empty values to the SQL string. This means that the problem is not limited to the `bildteil` field, but that no value can be deleted from any form field for now. 

I could solve this by either
- remove the check for empty values and allow empty values in all fields
- allow empty values for a specific set of field names (or only for `bildteil`)

**Comment by gioele** [comment]
That line of code is there since the very first import of the original code, either this error is caused by something else or it has always been like that (improbable).

**Comment by gioele** [comment]
I'd blame the code around line 48 or some later interactions with its effects

```php
	//Datenbankfelder identifizieren:
	if (mb_substr($field["key"],0,6)!="field_") {
		unset($felder[$field["key"]]);
	}
	//Nullwerte als NULL eintragen
	if ($field["value"]="") {
		$field["value"]="NULL";
	}
```

**Comment by gioele** [comment]
Fixed in commit e691e06dfe978b47c83621099e67834cf747136f.

By the way, it is very likely that the existing code never updated any field, as it was unconditionally overwriting the value of the field to be removed:

   `if ($field["value"]="") {`

vs.

   `if ($field["value"] == "") {`
